### DISCUSSION TITLE
How static are the parameters for a tool?

### DISCUSSION THREAD

**Discussion by abhi18av** [TORCH-Consortium/MAGMA#7] [Q&A] [answered]
Hi @TimHHH, @adippenaar  and @LennertVerboven ,

I was wondering whether you could help me understand whether the non-file arguments used within a module (for a tool) are static and are not expected to be changed frequently?

For example, 

In the command below, do you expect to change the `COUNT_UNPAIRED`, `READ_LENGTH `, `COVERAGE_CAP` very often while running the pipeline?

```nextflow
    $JAVA -Xmx64G -jar $GATK CollectWgsMetrics
    -R $REFERENCE
    -I $OUT_DIR/mapped/$SAMPLE_ID.recal_reads.bam
    --READ_LENGTH 0
    --COVERAGE_CAP 10000
    --COUNT_UNPAIRED
    -O $OUT_DIR/stats/$SAMPLE_ID.recal_reads.WgsMetrics
```

If you expect these to be changed/updated frequently, I'd rather encapsulate them in a `PARAMETER`  variable

```nextflow 
   def PARAMETERS =  "  --READ_LENGTH 0  --COVERAGE_CAP 10000 --COUNT_UNPAIRED"

    $JAVA -Xmx64G -jar $GATK CollectWgsMetrics \\
    -R $REFERENCE \\
    -I $OUT_DIR/mapped/$SAMPLE_ID.recal_reads.bam \\
    ${PARAMETERS} \\
    -O $OUT_DIR/stats/$SAMPLE_ID.recal_reads.WgsMetrics

```

As of now I'm inclined towards the latter approach so that it's more future friendly - what are your thoughts?

**Comment by TimHHH** [comment, marked as answer]
Hey @abhi18av 
The ```COVERAGE_CAP``` and ```COUNT_UNPAIRED``` parameters are very unlikely to change in the future, the first just has to be very high to ensure that all coverage is counted and the second ensures that we do get coverage values even if we are using single read samples. The ```READ_LENGTH``` value could potentially be extracted from the sequence read data, but this is not a priority as the most important wgs stats are still calculated appropriately despite this value.
You are welcome to add these to a ```parameter``` variable though, because it may be interesting for other bacterial species, but I wouldn't want them to 'bloat' the default user interface.

**Reply by abhi18av** [reply]
Ah, okay. This is helpful @TimHHH !

Therefore, to avoid the `bloat` issue - what I'll do is that I'll put this as the sensible default and if the user want to change this they can add this parameter while invoking the pipeline.

**Comment by TimHHH** [comment]
PS we should definitely use this approach for the ```VariantRecalibrator``` tool during VQSR.
